spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1430,6 +1430,27 @@ object QueryPlanSerde extends Logging with CometExprShim {
         val optExpr = scalarFunctionExprToProto("ascii", childExpr)
         optExprWithInfo(optExpr, expr, castExpr)
 
+      case RegExpReplace(subject, pattern, replacement, startPosition) =>
```

Comment:
Could you also check that `COMET_REGEXP_ALLOW_INCOMPATIBLE` is enabled? There are differences in behavior between Java's and Rust's regex implementation.

Reply:
`RLike` has this behavior:

```scala
      case RLike(left, right) =>
        // we currently only support scalar regex patterns
        right match {
          case Literal(pattern, DataTypes.StringType) =>
            if (!RegExp.isSupportedPattern(pattern.toString) &&
              !CometConf.COMET_REGEXP_ALLOW_INCOMPATIBLE.get()) {
              withInfo(
                expr,
                s"Regexp pattern $pattern is not compatible with Spark. " +
                  s"Set ${CometConf.COMET_REGEXP_ALLOW_INCOMPATIBLE.key}=true " +
                  "to allow it anyway.")
              return None
            }
```